test/test_distributions.py

```diff
@@ -270,3 +273,78 @@ def test_discrete_with_logits(jax_dist, dist_args):
     expected_pmf = jax_dist(*logit_args, is_logits=True).logpmf(actual_sample)
     assert_allclose(actual_pmf, expected_pmf, rtol=1e-6)
+
+
+########################################
+# Tests for constraints and transforms #
+########################################
+
+
+@pytest.mark.parametrize('constraint, x, expected', [
+    (constraints.greater_than(1), 3, True),
+    (constraints.greater_than(1), np.array([-1, 1, 5]), np.array([False, False, True])),
+    (constraints.integer_interval(-3, 5), 0, True),
+    (constraints.integer_interval(-3, 5), np.array([-5, -3, 0, 1.1, 5, 7]),
+     np.array([False, True, True, False, True, False])),
+    (constraints.interval(-3, 5), 0, True),
+    (constraints.interval(-3, 5), np.array([-5, -3, 0, 5, 7]),
+     np.array([False, False, True, False, False])),
+    (constraints.positive, 3, True),
+    (constraints.positive, np.array([-1, 0, 5]), np.array([False, False, True])),
+    (constraints.real, -1, True),
+    (constraints.real, np.array([np.inf, np.NINF, np.nan, np.pi]),
+     np.array([False, False, False, True])),
+    (constraints.simplex, np.array([0.1, 0.3, 0.6]), True),
+    (constraints.simplex, np.array([[0.1, 0.3, 0.6], [-0.1, 0.6, 0.5], [0.1, 0.6, 0.5]]),
+     np.array([True, False, False])),
+    (constraints.unit_interval, 0.1, True),
+    (constraints.unit_interval, np.array([-5, 0, 0.5, 1, 7]),
+     np.array([False, False, True, False, False])),
+], ids=idfn)
+def test_constraints(constraint, x, expected):
+    assert_array_equal(constraint(x), expected)
+
+
+@pytest.mark.parametrize('shape', [(), (1,), (3,), (5,), (3, 1), (1, 3), (5, 3)], ids=idfn)
+@pytest.mark.parametrize('constraint', [
+    constraints.greater_than(2),
+    constraints.interval(-3, 5),
+    constraints.positive,
+    constraints.real,
+    constraints.simplex,
+    constraints.unit_interval,
+], ids=idfn)
+def test_biject_to(constraint, shape):
+    if constraint is constraints.simplex and not shape:
+        return
+    transform = biject_to(constraint)
+    rng = random.PRNGKey(0)
+    x = random.normal(rng, shape)
+    y = transform(x)
+
+    # test codomain
+    batch_shape = shape if transform.event_dim == 0 else shape[:-1]
+    assert_array_equal(transform.codomain(y), np.ones(batch_shape))
+
+    # test inv
+    z = transform.inv(y)
+    assert_allclose(x, z, atol=1e-6)
+
+    # test domain, currently all is constraints.real
+    assert_array_equal(transform.domain(z), np.ones(shape))
+
+    # test log_abs_det_jacobian
+    actual = transform.log_abs_det_jacobian(x, y)
+    assert np.shape(actual) == batch_shape
+    if len(shape) == transform.event_dim:
+        if constraint is constraints.simplex:
+            return
+            # TODO compare against numerical values when jvp rule for `cumsum`, `cumprod` is available
+            # expected = np.linalg.slogdet(jax.jacobian(transform)(x)[::-1, :])[1]
+            # inv_expected = np.linalg.slogdet(jax.jacobian(transform.inv)(y)[:, ::-1])[1]
```

Comment:
currently, jvp rule for cumsum is not available yet. I have compared the result against PyTorch.

Reply:
Hmm..seems like we'll need this for dirichlet, but it should't be that complicated to implement.

Reply:
Yes, we'll need to implement a custom primitive for this. For cumsum, it seems not complicated because the jacobian is simply a upper triangular matrix will all entries are 1. It is just a bit more tricky for cumprod. I will make an issue for this.